codec.go

```diff
@@ -278,6 +298,10 @@ func (c *Codec) encodePayload(ctx context.Context, payload *common.Payload) (*co
 	}
 	req.Header.Set("X-Temporal-Metadata", base64.StdEncoding.EncodeToString(md))
 
+	for header, values := range c.customHeaders {
+		req.Header.Set(header, strings.Join(values, ","))
```

Comment:
Calling `Header.Add` might be easier here. It will under the hood take care of joining the headers I believe.

Is there are restriction in the value? What if it contains a ','?

Reply:
Good catch I  completely forgot that `Add` exists for some reason 😵 .
I didn't add restriction, but after your comment I added some url sanitization for the value so that users can add whatever they want.